app/bundles/CoreBundle/Tests/Functional/Field/Notification/CustomFieldNotificationFunctionalTest.php

```diff
@@ -0,0 +1,72 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Mautic\CoreBundle\Tests\Functional\Field\Notification;
+
+use Mautic\CoreBundle\Entity\Notification;
+use Mautic\CoreBundle\Entity\NotificationRepository;
+use Mautic\CoreBundle\Test\MauticMysqlTestCase;
+use Mautic\LeadBundle\Entity\LeadField;
+use Mautic\LeadBundle\Field\Notification\CustomFieldNotification;
+use Mautic\LeadBundle\Model\FieldModel;
+use Symfony\Contracts\Translation\TranslatorInterface;
+
+class CustomFieldNotificationFunctionalTest extends MauticMysqlTestCase
+{
+    protected $useCleanupRollback = false;
+    private TranslatorInterface $translator;
+    private CustomFieldNotification $notifier;
+    private LeadField $leadField;
+
+    protected function setUp(): void
+    {
+        parent::setUp();
+
+        $this->translator = $this->getContainer()->get('translator');
+        $this->notifier   = $this->getContainer()->get('mautic.lead.field.notification.custom_field');
+        $this->leadField  = $this->createCustomField();
+    }
+
+    public function testNoNotificationWhenLeadFieldCanNotUpdateForInvalidUser(): void
+    {
+        $this->notifier->customFieldCannotBeUpdated($this->leadField, -1);
+        /** @var NotificationRepository $notificationRepo */
+        $notificationRepo   = $this->em->getRepository(Notification::class);
```

Comment:
```suggestion
        $notificationRepo   = $this->em->getRepository(Notification::class);
        \assert($notificationRepo instanceof NotificationRepository);
```
Plus other occurrences in this diff